esp32 multiple boards samples fail to link with `undefined section 'tdata'`

**Describe the bug**
Multiple esp32 boards' tests and samples fail to link with ```undefined section `tdata'```
Affects at least: ttgo_t8c3, xiao_esp32c3, icev_wireless , stamp_c3, esp32c3_luatos_core, esp32c3_devkitm 

**To Reproduce**
For ex:
1. mkdir build; cd build
2. cmake -GNinja -DBOARD=ttgo_t8c3 ../tests/kernel/common/ -DCONFIG_MINIMAL_LIBC=y
3. ninja
4. See error

**Expected behavior**
no link errors

**Impact**
Many (hundredths) of CI build failures in the weekly CI run. Likely to trigger also in PRs CI.

**Logs and console output**
https://github.com/zephyrproject-rtos/zephyr/runs/26572698834
```
zephyr-sdk-0.16.8/riscv64-zephyr-elf/bin/../lib/gcc/riscv64-zephyr-elf/12.2.0/../../../../riscv64-zephyr-elf/bin/ld.bfd:zephyr/linker_zephyr_pre0.cmd:612: undefined section `tdata' referenced in expression
collect2: error: ld returned 1 exit status
```

**Environment (please complete the following information):**
 - OS: Linux CI
 - Toolchain (e.g Zephyr SDK, ...): 0.16.8
 - Commit SHA or Version used: Main as of now 18ebfe59d043f8b53bdd0c7ed48e2c9920abb881

**Additional context**
Issue introduced by b38d4300b9711c48fb62d37c2536e28004ef1858

--- a/soc/espressif/esp32c3/default.ld
+++ b/soc/espressif/esp32c3/default.ld
@@ -787,10 +787,12 @@ SECTIONS
   {
     . = ALIGN(4);
 
+#ifdef CONFIG_THREAD_LOCAL_STORAGE
      /* create explicit symbol for __tdata_start so that it is loaded
       * into proper DROM region atributted by AT keyword below
       */
     __tdata_start = ADDR(tdata);
+#endif
 
     _rodata_reserved_end = ABSOLUTE(.);
     _image_rodata_end = ABSOLUTE(.);

